### PR TITLE
CLOUDSTACK-10012: Embedded Jetty for CloudStack

### PR DESCRIPTION
**Pull request by rhtyd**
This is work-in-progress WIP embedded jetty support. Feedback welcome!

Changes adapted from PR #372 and #1546

Signed-off-by: Rohit Yadav <rohit.yadav@shapeblue.com>

### CODE REVIEW

**Comment by marcaurele** [comment]
@rhtyd What's the reason to create another (sub)module of the client, instead of adding a profile to the client module to either produce a tomcat or jetty artifact?

Would be nice to created a shaded jar including Jetty server, to have a "short" classpath (bouncycastle JCE is still required on the classpath, therefore it's not possible to create a single fat jar to run due to the signature checks). I could help you on that since it's the way we run it ourself.

**Comment by rhtyd** [comment]
@marcaurele thanks, any help would be great. The important goal is to move away from distro provided tomcat, I'm open to all possible ways to do that. Having a fat jar would be great, but for the first step if that's not possible I can do without that approach.

File: packaging/cloud-management.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+ . /etc/cloudstack/management/tomcat6.conf
+SERVLETENGINE='jetty'
+# For SELinux we need to use 'runuser' not 'su'
+if [ -x "/sbin/runuser" ]; then
+    SU="/sbin/runuser -s /bin/sh"
+else
+    SU="/bin/su -s /bin/sh"
+fi
```
Comment (wido) [inline packaging/cloud-management.sh]: You don't seem to use this?

File: packaging/cloud-management.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# Define the tomcat log file
+CLOUD_LOG="${CLOUD_LOG:-/var/log/${NAME}-initd.log}"
+
+# Define the pid file name
+# If change is needed, use sysconfig instead of here
+export CLOUD_PID="${CLOUD_PID:-/var/run/${NAME}.pid}"
+CLASSPATH="${CLASSPATH}:${CLOUD_HOME}/lib/jetty-runner.jar"
+CLASS="org.eclipse.jetty.runner.Runner --classes /etc/cloudstack/management --path /client $CLOUD_HOME/webapps/client  --stop-port 8888 --stop-key monkeystop"
+
+if [ "$1" = "start" ]; then
```
Comment (wido) [inline packaging/cloud-management.sh]: Shouldn't this just be a wrapper script that systemd can execute? As we want to start/stop everything through systemd

**Comment by rhtyd** [comment]
@wido thanks, this does not work as of now and needs some further changes. I wanted to start and do this in open to gather some initial feedback and advise.

**Comment by marcaurele** [comment]
@rhtyd How could we work together on a branch? My idea was to push what we have (still based on 4.4.2) and let you adapt it to make it work with upstream version since I don't have a way to test upstream correctly

**Comment by rhtyd** [comment]
Thanks @marcaurele if you can share your idea as a PR, I can use it as reference and adapt as necessary. I think the general goal is to fix the problem, whatever way necessary. By moving away from tomcat (distro provided), can help in CloudStack's adoption (and as a fatjar somewhere in future).

**Comment by rhtyd** [comment]
Closing this in light in #2207

**Pull request closed without merging**